realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -481,6 +495,17 @@ public Builder rxFactory(RxObservableFactory factory) {
             return this;
         }
 
+        /**
+         * Sets the initial data in {@link io.realm.Realm}. This transaction will be executed only for the first time
+         * when database file is created or while migrating the data when {@link Builder#deleteRealmIfMigrationNeeded()} is set.
+         *
+         * @param transaction transaction to execute.
+         */
+        public Builder initialData(Realm.Transaction transaction) {
```

Comment:
I don't really think user really need `public Builder initialData(DynamicRealm.Transaction transaction);`, by supplying that, it also coupled the typed Realm and dynamic Realm even more. We also have to consider a right place to trigger the initial dynamic Realm transaction when user only uses dynamic Realm. So i prefer:
1. Only initial transaction for typed Realm
2. It will only be triggered when first time create Realm (version == unversioned)